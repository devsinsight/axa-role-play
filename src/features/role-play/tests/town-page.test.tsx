import React from "react";
import renderer from "react-test-renderer";
import { pagination, slide, professions } from "../../../../test/mockData";
import ProfessionFilter from "../components/inhabitants-profession-filter";
import InhabitantsList from "../components/inhabitants-list";
import Friends from "../components/inhabitants-friends";
import Professions from "../components/inhabitants-professions";

it("should render ProfessionFilter component", () => {
  const professionFilter = renderer.create(
    <ProfessionFilter
      profession={"All"}
      professions={professions}
      handleChange={jest.fn()}
    />
  );

  expect(professionFilter).toMatchSnapshot();
});

it("should render InhabitantsList component when transition 'in' is false", () => {
  const inhabitantsList = renderer.create(
    <InhabitantsList
      pagination={pagination}
      slide={slide}
      next={jest.fn()}
      prev={jest.fn()}
    />
  );

  expect(inhabitantsList).toMatchSnapshot();
});

it("should render gnomes friends", () => {
  const inhabitantsList = renderer.create(<Friends friends={["cat", "dog"]} />);

  expect(inhabitantsList).toMatchSnapshot();
});

it("should render gnomes profressions", () => {
  const inhabitantsList = renderer.create(
    <Professions professions={professions} />
  );

  expect(inhabitantsList).toMatchSnapshot();
});
