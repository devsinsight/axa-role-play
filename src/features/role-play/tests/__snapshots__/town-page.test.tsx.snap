// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InhabitantsList component when transition 'in' is false 1`] = `<div />`;

exports[`should render ProfessionFilter component 1`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
    data-shrink={true}
    id="demo-simple-select-filled-label"
  >
    Professions
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="demo-simple-select-label demo-simple-select"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
      id="demo-simple-select"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      All
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
      value="All"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`should render gnomes friends 1`] = `
<span>
  My friends:
  <span>
     
    cat
    ,
  </span>
  <span>
     
    dog
    .
  </span>
</span>
`;

exports[`should render gnomes profressions 1`] = `
<span>
  My professions:
  <span>
     
    All
    ,
  </span>
  <span>
     
    Cook
    ,
  </span>
  <span>
     
    Smelter
    ,
  </span>
  <span>
     
    Tinker
    ,
  </span>
  <span>
     
    Carpenter
    ,
  </span>
  <span>
     
    Farmer
    ,
  </span>
  <span>
     
    Stonecarver
    ,
  </span>
  <span>
     
    Brewer
    ,
  </span>
  <span>
     
    Tax inspector
    ,
  </span>
  <span>
     
    Prospector
    ,
  </span>
  <span>
     
    Baker
    ,
  </span>
  <span>
     
    Miner
    ,
  </span>
  <span>
     
    Medic
    ,
  </span>
  <span>
     
    Gemcutter
    ,
  </span>
  <span>
     
    Mason
    ,
  </span>
  <span>
     
    Tailor
    ,
  </span>
  <span>
     
    Metalworker
    ,
  </span>
  <span>
     
    Woodcarver
    ,
  </span>
  <span>
     
    Potter
    .
  </span>
</span>
`;
